sparkth_mcp/canvas/tools.py

```diff
@@ -0,0 +1,642 @@
+    ModulePayload,
+    PagePayload,
+    PageRequest,
+    QuestionPayload,
+    QuestionRequest,
+    QuizPayload,
+    QuizRequest,
+    UpdateModuleItemPayload,
+    UpdateModulePayload,
+    UpdatePagePayload,
+    UpdateQuestionPayload,
+    UpdateQuizPayload,
+)
+
+
+@mcp.tool
+async def canvas_authenticate(api_url: str, api_token: str) -> json:
+    """
+    Authenticate the provided Canvas API URL and token.
+    If either argument is missing, the client must supply it. Default values for required fields are never assumed.
+
+    Args:
+        api_url (str): The Canvas API base URL (e.g. https://canvas.instructure.com/api/v1/).
+        api_token (str): The user's Canvas API token used for authentication.
+    """
+    res = {}
+
+    try:
+        res = await CanvasClient.authenticate(api_url, api_token)
+    except AuthenticationError as e:
+        return {"status": e.status_code, "message": e.message}
+
+    return {"status": res}
+
+
+@mcp.tool
+async def canvas_get_courses(api_url: str, api_token: str) -> json:
+    """
+    Retrieve all courses for the user.
+
+    If either argument is missing, the client must provide it. Default values for required fields are never assumed.
+    If the credentials have not already been authenticated, they must be validated before
+    retrieving the course list.
+
+    Args:
+        api_url (str): The Canvas API base URL (e.g., https://canvas.instructure.com/api/v1/).
+        api_token (str): The user's Canvas API token used for authentication.
+    """
+    canvas_client = CanvasClient(api_url, api_token)
+    return await canvas_client.request_bearer("GET", "courses")
+
+
+@mcp.tool
+async def canvas_get_course(api_url: str, api_token: str, course_id) -> json:
+    """
+    Retrieve a single course for the user by course_id.
+
+    If any argument is missing, the client must provide it. Default values for required fields are never assumed.
+    If the credentials have not already been authenticated, they must be validated before
+    retrieving the course list.
+
+    Args:
+        api_url (str): The Canvas API base URL (e.g., https://canvas.instructure.com/api/v1/).
+        api_token (str): The user's Canvas API token used for authentication.
+        course_id (int): The id for the course to be retrieved
+    """
+    canvas_client = CanvasClient(api_url, api_token)
+    return await canvas_client.request_bearer("GET", f"courses/{course_id}")
+
+
+@mcp.tool
+async def canvas_create_course(payload: CoursePayload) -> json:
+    """
+    Create a new course on Canvas.
+
+    If any argument is missing, the client must provide it. Default values for required fields are never assumed.
+    If the credentials have not already been authenticated, they must be validated before
+    retrieving the course list.
+
+    Args:
+        api_url (str): The Canvas API base URL (e.g., https://canvas.instructure.com/api/v1/).
+        api_token (str): The user's Canvas API token used for authentication.
```

Comment:
Docstring lists `api_url` and `api_token` as args but the function only takes payload.